packages/connectivity/src/http-agent/http-agent.ts

```diff
@@ -119,7 +119,11 @@ function getTrustStoreOptions(destination: Destination): Record<string, any> {
  * @returns Options, which can be used later by tls.createSecureContext() e.g. pfx and passphrase or an empty object, if the protocol is not 'https:' or no client information are in the definition.
  */
 function getKeyStoreOption(destination: Destination): Record<string, any> {
-  if (destination.keyStoreName && destination.keyStorePassword) {
+  if (
+    destination.keyStoreName &&
+    destination.keyStorePassword &&
+    destination.authentication === 'ClientCertificateAuthentication'
```

Comment:
[nth] Perhaps add a link to the original issue or one or two sentences to explain that the keyStore property are present also for the desintaiton service to sign SamlAssertion but only for the `ClientCertificateAuthentication` we as the SDK need to consider it.

Reply:
not only; please note that from now on all the destinations authentication types where a token is fetched via a token issuance endpoint do support explicit mTLS keystore dedicated to token fetching only...we have agreed this will be managed as a feature request going forward...

Reply:
@ptesny Yes, we talked about this, but I later realized, that this is nothing that we need to do. I believe the destination service is already doing that for us: https://help.sap.com/docs/CP_CONNECTIVITY/cca91383641e40ffbe03bdc78f00f681/2c162aaa9e464480b2307879d1b865f5.html?q=mtls

In the client secret case, we are not doing anything either.

Reply:
@marikaner , indeed, that's correct; thus no further action on your side; many thanks